packages/sdk/src/contracts/layer-zero-module.contract.ts

```diff
@@ -0,0 +1,193 @@
+import {Network} from '@holographxyz/networks'
+import {Address, ExtractAbiFunctionNames} from 'abitype'
+import {isCallException} from 'ethers'
+
+import {HolographByNetworksResponse, getSelectedNetworks, mapReturnType} from '../utils/contracts'
+import {getContract} from '../utils/abitype'
+import {ContractRevertError, EthersError, HolographError} from '../errors'
+import {Providers, HolographLogger, Config} from '../services'
+import {LayerZeroModuleABI} from '../constants/abi/develop'
+import {Addresses} from '../constants/addresses'
+
+type LayerZeroModuleFunctionNames = ExtractAbiFunctionNames<typeof LayerZeroModuleABI, 'view'>
+
+/**
+ * @group Contracts
+ * LayerZeroModule
+ *
+ * @remarks
+ *
+ * Holograph module for enabling LayerZero cross-chain messaging.
+ * This contract abstracts all of the LayerZero specific logic into an isolated module.
+ *
+ */
+export class LayerZeroModule {
+  /** The list of networks in which the contract was instantiated. */
+  public readonly networks: Network[]
+  /** The record of addresses per chainId. */
+  private readonly _addresses: Record<number, Address> = {}
+  private readonly _providers: Providers
+  private _logger: HolographLogger
+
+  constructor(private readonly config: Config, parentLogger?: HolographLogger) {
+    this._providers = new Providers(config)
+
+    if (parentLogger) {
+      this._logger = parentLogger.addContext({className: LayerZeroModule.name})
+    } else {
+      this._logger = HolographLogger.createLogger({className: LayerZeroModule.name})
+    }
+
+    this.networks = this.config.networks
+  }
+
+  /**
+   * @readonly
+   * Get  the LayerZeroModule contract address according to environment and chainId module.
+   * @param chainId The chainId of the network to get the result from.
+   * @returns The LayerZeroModule contract address in the provided network.
+   */
+  getAddress(chainId?: number | string): Address {
+    return Addresses.layerZeroModule(this.config.environment, Number(chainId)) as Address
+  }
+
+  private async _getContractFunction(chainId: number, functionName: LayerZeroModuleFunctionNames, ...args: any[]) {
+    const logger = this._logger.addContext({functionName})
+    const provider = this._providers.byChainId(chainId)
+    const address = this.getAddress(chainId)
+
+    const contract = getContract({address, abi: LayerZeroModuleABI, signerOrProvider: provider})
+
+    let result
+    try {
+      result = await contract[functionName](...args)
+    } catch (error: any) {
+      let holographError: HolographError
+
+      if (isCallException(error)) {
+        holographError = new ContractRevertError('LayerZeroModule', functionName, error)
+      } else {
+        holographError = new EthersError(error, functionName)
+      }
+
+      logger.logHolographError(error)
+
+      throw holographError
+    }
+    return mapReturnType(result)
+  }
+
+  /**** getGasParametersByNetworks ****/
```

Comment:
nit: I don't think these functions need these dividers with the function name above. imo just kind of redundant

Reply:
@feliperodriguess I agree with @alexanderattar, we can remove the extra comment. I can see that its a little inconsistent at the moment. So just remove it from your current files and i think we can merge it in.

Reply:
I just followed the existing pattern, but I do agree w/ you guys

I've just removed that from all contract files